pkgs/development/julia-modules/default.nix

```diff
@@ -0,0 +1,178 @@
+{ lib
+, callPackage
+, runCommand
+, fetchFromGitHub
+, fetchgit
+, fetchurl
+, git
+, makeWrapper
+, writeText
+, writeTextFile
+, python3
+, stdenv
+
+, julia
+}:
+
+{ extraLibs ? []
+, precompile ? true
+, setDefaultDepot ? true
+, makeWrapperArgs ? ""
+, packageOverrides ? {}
+}@args:
```

Comment:
Is there a reason for this nesting?
I thought overrides stack, such this could be part of the other arguments and be accessed using `julia.withPackages.override`.

Reply:
I thought it might be more idiomatic to separate the "dependencies" from the "arguments" of this function, and pass the latter as actual arguments. I don't feel strongly about it though.

Reply:
I don't have a strong feeling about it either. My two cents were that most people probably won't set any arguments and that without it the interface would be closer to the one for python.

Reply:
Okay, upon reflection I think it's true people won't often want to change the arguments, so it's cleaner to go back to the override-based API. Did this in 66d562450a0c8bf0a6e92d7404831937ee0596c3